nautobot/core/templates/inc/nav_menu.html

```diff
@@ -111,3 +112,73 @@
         </div>
     </div>
 </nav>
+
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+        var navbar = document.querySelector('.navbar-fixed-left');
+        var dropdownToggles = document.querySelectorAll('.navbar-fixed-left .navbar-nav > .dropdown > a[data-toggle="collapse"]');
+        var dropdowns = document.querySelectorAll('.navbar-fixed-left .navbar-nav .collapse');
+        var lastDropdownId = localStorage.getItem('lastOpenedDropdown');
+        var savedScrollPosition = localStorage.getItem('navbarScrollPosition');
+
+        // Debounce function
+        function debounce(func, wait, immediate) {
+            var timeout;
+            return function() {
+                var context = this, args = arguments;
+                var later = function() {
+                    timeout = null;
+                    if (!immediate) func.apply(context, args);
+                };
+                var callNow = immediate && !timeout;
+                clearTimeout(timeout);
+                timeout = setTimeout(later, wait);
+                if (callNow) func.apply(context, args);
+            };
+        }
+
+        // Close all dropdowns except the one specified
+        function closeAllDropdownsExcept(exceptId) {
+            dropdowns.forEach(function(collapse) {
+                if (collapse.id !== exceptId && collapse.classList.contains('in')) {
+                    $(collapse).collapse('hide');
```

Comment:
Is this jQuery?

Reply:
This is a mix of Javascript and Bootstrap integrated Jquery.  I am leveraging the built-in collapse feature within Bootstrap.